src/menus/LabelMenus.cpp

```diff
@@ -177,13 +177,17 @@ using EditFunctionWithProgress =
 void EditByLabel(
    AudacityProject& project, TrackList& tracks,
    const SelectedRegion& selectedRegion, EditFunctionWithProgress action,
-   std::unique_ptr<BasicUI::ProgressDialog> progress)
+   ProgressReporter progress)
 {
    Regions regions;
 
    GetRegionsByLabel(tracks, selectedRegion, regions);
    if (regions.empty())
       return;
+   // Apply action on tracks starting from
+   // labeled regions in the end. This is to correctly perform
+   // actions like 'Delete' which collapse the track area.
+   std::reverse(regions.begin(), regions.end());
```

Comment:
Okay, this commit totally rewrites the previous.  But it does it with a very weird and unnatural inversion of control.  What led to this choice?

I think it would make more understandable code if instead you had a factory making a function object that you just call to update the user, and you write loops more naturally, just bumping the count captured by reference in the factory.

Did you not do this just because you could not figure out the lambda capture?

Abstracting just the test for user cancellation means decisions about how to iterate the range aren't forced -- and you would not need the line above but could iterate with `rbegin` and `rend` as it was.

Really I don't like this.  Try it as I suggested.